app/server/appsmith-plugins/googleSheetsPlugin/src/main/java/com/external/config/AppendMethod.java

```diff
@@ -2,38 +2,201 @@
+            rowObjectListFromBody = this.getRowObjectListFromBody(this.objectMapper.readTree(body));
+        } catch (JsonProcessingException e) {
+            // Should never enter here
         }
+        assert rowObjectListFromBody != null;
+        RowObject rowObjectFromBody = rowObjectListFromBody.get(0);
+        assert rowObjectFromBody != null;
+        final String row = String.valueOf(
+                Integer.parseInt(methodConfig.getTableHeaderIndex()) + rowObjectFromBody.getCurrentRowIndex());
+        final MethodConfig newMethodConfig = methodConfig
+                .toBuilder()
+                .queryFormat("ROWS")
+                .spreadsheetRange(row)
+                .rowLimit("1")
+                .build();
+
+        getValuesMethod.validateMethodRequest(newMethodConfig, body);
+
+        List<RowObject> finalRowObjectListFromBody = rowObjectListFromBody;
+        return getValuesMethod
+                .getClient(client, newMethodConfig, body)
+                .headers(headers -> headers.set(
+                        "Authorization",
+                        "Bearer " + oauth2.getAuthenticationResponse().getToken()))
+                .exchange()
+                .flatMap(clientResponse -> clientResponse.toEntity(byte[].class))
+                .map(response -> {// Choose body depending on response status
+                    byte[] responseBody = response.getBody();
+
+                    if (responseBody == null || !response.getStatusCode().is2xxSuccessful()) {
+                        return Mono.error(new AppsmithPluginException(
+                                AppsmithPluginError.PLUGIN_ERROR,
+                                "Could not map request back to existing data"));
+                    }
+                    String jsonBody = new String(responseBody);
+                    JsonNode jsonNodeBody = null;
+                    try {
+                        jsonNodeBody = objectMapper.readTree(jsonBody);
+                    } catch (IOException e) {
+                        Mono.error(new AppsmithPluginException(
+                                AppsmithPluginError.PLUGIN_JSON_PARSE_ERROR,
+                                new String(responseBody),
+                                e.getMessage()
+                        ));
+                    }
+
+                    // This is the object with the original values in the referred row
+                    final JsonNode jsonNode = getValuesMethod
+                            .transformResponse(jsonNodeBody, methodConfig);
+
+                    // This is the rowObject for original values
+                    final RowObject returnedRowObject = this.getRowObjectListFromBody(jsonNode).get(0);
+
+                    // We replace these original values with new ones
+                    finalRowObjectListFromBody.forEach(rowObject -> {
+                        final LinkedHashMap<String, String> valueMap = new LinkedHashMap<>(returnedRowObject.getValueMap());
```

Comment:
```suggestion
                        final Map<String, String> valueMap = new LinkedHashMap<>(returnedRowObject.getValueMap());
```

We should typically use interfaces to define the variable. The actual implementation will be determined by the initialization.